test/behavior/asm.zig

```diff
@@ -150,12 +150,28 @@ test "struct/array/union types as input values" {
     ); // fails
     asm volatile (""
         :
-        : [_] "m" (@as(struct { x: u32, y: u8 }, undefined)),
+        : [_] "m" (@as(packed struct { x: u32, y: u8 }, undefined)),
     ); // fails
+    // TODO: CBE struggles with unions here
+    if (builtin.zig_backend == .stage2_c)
+        return error.SkipZigTest
+    else
+        asm volatile (""
+            :
+            : [_] "m" (@as(packed union { x: u32, y: u8 }, undefined)),
+        ); // fails
```

Comment:
Wait, with the `asm` in the `else` branch, shouldn't this just be a nop in the CBE case rather than an early return? Put differently, why is this not:
```suggestion
    if (builtin.zig_backend != .stage2_c)
        asm volatile (""
            :
            : [_] "m" (@as(packed union { x: u32, y: u8 }, undefined)),
        ); // fails
```

Reply:
This is temporary so I didn't bother spending a lot of thought on it. Suggestion applied.